src/lib.rs

```diff
@@ -844,6 +879,214 @@ pub unsafe trait FromBytes: FromZeroes {
     }
 }
 
+/// Types whose validity can be checked at runtime, allowing them to be
+/// conditionally converted from byte slices.
+///
+/// WARNING: Do not implement this trait yourself! Instead, use
+/// `#[derive(TryFromBytes)]`.
+///
+/// `TryFromBytes` types can safely be deserialized from an untrusted sequence
+/// of bytes by performing a runtime check that the byte sequence contains a
+/// valid instance of `Self`.
+///
+/// `TryFromBytes` is ignorant of byte order. For byte order-aware types, see
+/// the [`byteorder`] module.
+///
+/// # Safety
+///
+/// On its own, `T: TryFromBytes` does not make any guarantees about the layout
+/// or representation of `T`. It merely provides the ability to perform a
+/// validity check at runtime via methods like [`try_from_ref`].
+///
+/// TODO(#352): UnsafeCell? We need to specify that it's unsound to implement
+/// `TryFromBytes` for types which contain them, but we also don't want to allow
+/// code to use `TryFromBytes` to mean "has no `UnsafeCell`" in case we relax
+/// that requirement in the future.
```

Comment:
Need to address this before merging.

Reply:
Maybe, given the following warning about not providing stability guarantees and about making code silently become unsound, we can just omit mentioning this until we address #352 as a whole.